components/dashboard/src/start/CreateWorkspace.tsx

```diff
@@ -150,5 +150,59 @@ export class CreateWorkspace extends React.Component<CreateWorkspaceProps, Creat
       </div>}
     </StartPage>;
   }
+}
+
+export interface PrebuildInProgressProps {
+  runningPrebuild: {
+    prebuildID: string
+    workspaceID: string
+    starting: RunningWorkspacePrebuildStarting
+    sameCluster: boolean
+  };
+  onIgnorePrebuild: () => void;
+  onPrebuildSucceeded: () => void;
+}
+
+export function PrebuildInProgress(props: PrebuildInProgressProps) {
+  const logsEmitter = new EventEmitter();
+  const service = getGitpodService();
+  let pollTimeout: NodeJS.Timeout | undefined;
+
+  useEffect(() => {
+    const pollIsPrebuildDone = async () => {
+      const available = await service.server.isPrebuildDone(props.runningPrebuild.prebuildID);
+      if (available) {
+        props.onPrebuildSucceeded();
+        return;
+      }
+      pollTimeout = setTimeout(pollIsPrebuildDone, 10000);
+    }
+    // function watchPrebuild() {
+      service.server.watchHeadlessWorkspaceLogs(props.runningPrebuild.workspaceID);
+      pollIsPrebuildDone();
+    // }
+    // watchPrebuild();
+
+    const toDispose = service.registerClient({
+      // notifyDidOpenConnection: () => watchPrebuild(),
```

Comment:
@AlexTugarev Not sure how WebSocket reconnects work. Do you know if it's (still) necessary to call `service.server.watchHeadlessWorkspaceLogs()` again in `notifyDidOpenConnection()`?

Reply:
yes it's necessary, indeed.